tests/publiccloud/ipa.pm

```diff
@@ -36,9 +36,18 @@ sub run {
         my $system_max_boot_time = 120;
         my $out                  = script_output('grep "^Startup finished in" ' . $ipa->{logfile});
         record_info('Startup time', $out);
-        die 'Fail to find boot time in log' unless $out =~ /Startup finished in (\d{1,4}\.\d{3})s \(kernel\) \+ \d{1,4}\.\d{3}s \(initrd\) \+ \d{1,4}\.\d{3}s \(userspace\) = (\d{1,4}\.\d{3})s/;
-        record_info('Kernel boot is too slow',         result => 'fail') if $1 > $kernel_max_boot_time;
-        record_info('Overall system boot is too slow', result => 'fail') if $2 > $system_max_boot_time;
+        my $kernel_real_boot_time, $system_real_boot_time;
+        if ($out =~ /\= \d{1,2}min/) {
+            die 'Fail to find boot time in log' unless $out =~ /Startup finished in (\d{1,4}\.\d{3})s \(kernel\) \+ \d{1,4}\.\d{3}s \(initrd\) \+ \d{1,4}\.\d{3}s \(userspace\) = (\d{1,2})min (\d{1,4}\.\d{3})s/;
```

Comment:
You can use qr/foo/ to make the common parts of your regex a bit simpler. since you're looking already for the same pattern few times already. and you could replace the common parts with: `((\d+? min)? \d+\.?\d+s)` notice the space. 
give it a look [here](https://gist.github.com/foursixnine/5461d59877ae86d0ef8977df1b82758e)

Perhaps if you name the capture group also, you might end up with nice ` %+` [for magic variables](https://perldoc.perl.org/perlvar.html#Variables-related-to-regular-expressions)